src/compute/src/compute_state.rs

```diff
@@ -476,6 +476,7 @@ impl<'a, A: Allocate + 'static> ActiveComputeState<'a, A> {
                 export_ids = %dataflow.display_export_ids(),
                 as_of = ?as_of.elements(),
                 dataflow_expiration = ?dataflow_expiration.elements(),
+                dataflow_expiration_selection = ?expiration_selection,
```

Comment:
I think the easier (and more scalable) approach is to log all parameters that go into `expire_dataflow_at`, instead of adding an enum that might not have full coverage. Optionally, this can be within `expire_dataflow_at` at a `debug` level, so it doesn't spam everyone, but needs to be specifically enabled.